src/attr/_make.py

```diff
@@ -599,7 +599,7 @@ def _transform_attrs(
             had_default = True
 
     if field_transformer is not None:
-        attrs = field_transformer(cls, attrs)
+        attrs = AttrsClass(field_transformer(cls, attrs))
```

Comment:
This could be moved somewhere above this line, https://github.com/python-attrs/attrs/pull/824/files#diff-ca2708dc9bf685ad676dfaa594561d8fc5c18ae39a7a5bc517049ef3fce09541R584 ,
so you won't wrap attrs in class two times

```python
attrs_list = base_attrs + own_attrs
if field_transformer is not None:
        attrs_list = field_transformer(cls, attrs_list)
        
attrs = AttrsClass(attrs_list)
```